src/ECER.Managers.Registry/ApplicationHandlers.cs

```diff
@@ -46,6 +47,42 @@ public static async Task<string> Handle(SaveDraftApplicationCommand cmd, IApplic
     return applicationId;
   }
 
+  /// <summary>
+  /// Handles submitting a new application use case
+  /// </summary>
+  /// <param name="cmd">The command</param>
+  /// <param name="applicationRepository">DI service</param>
+  /// <param name="validationEngine">validationEngine</param>
+  /// <param name="mapper">DI service</param>
+  /// <returns></returns>
+  public static async Task<ApplicationSubmissionResult> Handle(SubmitApplicationCommand cmd, IApplicationSubmissionValidationEngine validationEngine, IApplicationRepository applicationRepository, IMapper mapper)
+  {
+    ArgumentNullException.ThrowIfNull(applicationRepository);
+    ArgumentNullException.ThrowIfNull(mapper);
+    ArgumentNullException.ThrowIfNull(cmd);
+
+    var applications = await applicationRepository.Query(new ApplicationQuery
+    {
+      ById = cmd.applicationId,
+      ByStatus = new Resources.Documents.Applications.ApplicationStatus[] { Resources.Documents.Applications.ApplicationStatus.Draft }
+    });
+
+    var draftApplicationResults = new ApplicationsQueryResults(mapper.Map<IEnumerable<Contract.Applications.Application>>(applications)!);
+    var draftApplication = draftApplicationResults.Items.FirstOrDefault();
```

Comment:
If the assumption there's only 1 draft, then it is ok to validate this as well - firstordefault is dangerous as it makes assumptions that may not be true